### ISSUE TITLE
Crash with Rads

### ISSUE THREAD

**Issue by Fortist** [turbodiesel4598/NuclearCraft#61]
`````
----` Minecraft Crash Report ----
// This doesn't make any sense!

Time: 9/12/16 11:06 PM
Description: Exception ticking world entities

java.lang.IllegalArgumentException: Duplicate id value for 29!
	at net.minecraft.entity.DataWatcher.func_75682_a(SourceFile:64)
	at nc.player.PlayerRads.<init>(PlayerRads.java:33)
	at nc.player.PlayerRads.register(PlayerRads.java:37)
	at nc.handler.RadHandler.onEntityConstructing(RadHandler.java:26)
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_1468_RadHandler_onEntityConstructing_EntityConstructing.invoke(.dynamic)
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54)
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140)
	at net.minecraft.entity.Entity.<init>(Entity.java:179)
	at net.minecraft.entity.EntityLivingBase.<init>(EntityLivingBase.java:119)
	at net.minecraft.entity.player.EntityPlayer.<init>(EntityPlayer.java:141)
	at net.minecraft.entity.player.EntityPlayerMP.<init>(EntityPlayerMP.java:138)
	at net.minecraftforge.common.util.FakePlayer.<init>(FakePlayer.java:22)
	at cofh.core.entity.CoFHFakePlayer.<init>(CoFHFakePlayer.java:29)
	at cofh.thermalexpansion.block.device.TileActivator.cofh_validate(TileActivator.java:99)
	at net.minecraft.world.World.func_72939_s(World.java:50532)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:489)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:636)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:186)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:396)
	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at net.minecraft.entity.DataWatcher.func_75682_a(SourceFile:64)
	at nc.player.PlayerRads.<init>(PlayerRads.java:33)
	at nc.player.PlayerRads.register(PlayerRads.java:37)
	at nc.handler.RadHandler.onEntityConstructing(RadHandler.java:26)
	at cpw.mods.fml.common.eventhandler.ASMEventHandler_1468_RadHandler_onEntityConstructing_EntityConstructing.invoke(.dynamic)
	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54)
	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140)
	at net.minecraft.entity.Entity.<init>(Entity.java:179)
	at net.minecraft.entity.EntityLivingBase.<init>(EntityLivingBase.java:119)
	at net.minecraft.entity.player.EntityPlayer.<init>(EntityPlayer.java:141)
	at net.minecraft.entity.player.EntityPlayerMP.<init>(EntityPlayerMP.java:138)
	at net.minecraftforge.common.util.FakePlayer.<init>(FakePlayer.java:22)
	at cofh.core.entity.CoFHFakePlayer.<init>(CoFHFakePlayer.java:29)
	at cofh.thermalexpansion.block.device.TileActivator.cofh_validate(TileActivator.java:99)
	at net.minecraft.world.World.func_72939_s(World.java:50532)
	at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:489)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:636)
```

**Comment by turbodiesel4598** [comment]
Thought this could be a problem. I've just uploaded [a new 1.9d](https://minecraft.curseforge.com/projects/nuclearcraft-mod/files/2329936), with some more config options for the data watcher values (I've defaulted the value that you found collided to 25 instead of 29, so try that first). All you can do is try any combination of four numbers from 18 to 31 and hope that it doesn't crash, because there is no way I know of other than by crashing to see which values are already being used.

**Comment by Fortist** [comment]
Well, 1 didnt work, on my test instance. Ill flip through that first and then try it on my actual instance.

**Comment by turbodiesel4598** [comment]
Well, let me just explain how confused I am right now :P 
I was told that the values had to be different, between 1 and 31, and that values 1 to 18 were taken up (or, most of them, at least) by vanilla. However, I set all of them to 1, and my client worked... so I'm really not sure what's going on. Perhaps try some values in the 20s and see if it works...
It's very strange indeed.

**Comment by Fortist** [comment]
WELL, I made each one different, and the problem was the first value being LastRads. So finally changing that to the next value (5), and my world loaded, also tested how radiation worked. Which reminds me, you shouldnt have all the sounds tied to MASTER. Blocks to blocks, some ambient sounds to ambient. I'd rather not have a heart attack every time I play with a nuke. 
Next Im going to see if all the values can be 5, and test from there. Otherwise, I guess Im fine.

**Comment by Fortist** [comment]
Okay, more testing. I cant have them all the same value or i'd get a duplicate ID crash.
Also, is creative blocking me from using Radaway/X or is it not seeing the rad value due to having different IDs?

**Comment by turbodiesel4598** [comment]
Oh yes, I need to fix the sound for 1.10.2 - thanks for reminding me.
I imagine there will be a few bugs, as this is more of a 'proof of concept' build. I'll say that this is a beta on curse.
